xmlrpc/man/tcospasswd.1
.TH tcosmd5 1 "Jun 21, 2006" "tcosmd5 man page"

.SH NAME
tcospasswd \- TCOS util to generate crytp TCOS passwords

.SH SYNOPSIS
.B tcospasswd
.RI [ options ]
.br

.SH DESCRIPTION

This manpage show options for 
.B tcospasswd
app.

.PP
\fBtcospasswd\fP is a small C app that create TCOS crypt 
of an string passed from cmdline.

.SH OPTIONS
.B
.IP clear_password
string to crypt (to shell scripts)

.B
.IP [NO_ARGUMENTS]
interactive mode

.SH EXAMPLES
.B
$ tcospasswd my_password

$ tcospasswd 

.SH BUGS
Contact Mario Izquierdo <mariodebian@gmail.com>

Homepage: http://www.tcosproject.org

.SH SEE ALSO
tcos.conf(5)

.SH AUTHOR
Mario Izquierdo (mariodebian) <mariodebian@gmail.com>


debian/changelog
initramfs-tools-tcos (0.84.5) unstable; urgency=low

  * Remove tcos-server-utils package (now contents are in tcos package)
  * Remove obsolete opengl dir
  * Remove obsolete discover dir
  * Remove obsolete esound dir
  * Remove old dir

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 24 Sep 2007 13:10:18 +0200

initramfs-tools-tcos (0.84.4) unstable; urgency=low

  * Make TCOS_EXTRA_MODULES new var in tcos.conf
  * gentcos:
    - Remove export TCOS_EXTRA_MODULES
    - Fix read -extramodules cmdline to allow more modules

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 24 Sep 2007 13:07:23 +0200

initramfs-tools-tcos (0.84.3) unstable; urgency=low

  * Only depend on mknbi on i386 arch
  * Patch debian/control in Ubuntu dapper with Source-Version

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 22 Sep 2007 21:23:44 +0200

initramfs-tools-tcos (0.84.2) unstable; urgency=low

  * bin/tcos-buildroot:
    - Fix typo in TCOS_CHROOT var
    - Add --gentcos option to build images with other params
  * Update Ubuntu gutsy kernel version
  * debian/control:
    - Replace source:version depends with 0.84.1
    - Fix some descriptions 

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 22 Sep 2007 18:52:55 +0200

initramfs-tools-tcos (0.84.1) unstable; urgency=low

  * Important move of code, some of tcosmonitor stuff move here:
    - udev
    - xmlrpc
    - lockscreen
    - hooks-addons
  * bin/tcos-buidchroot:
    - Script to build a (32bit) chroot to build tcos images

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri, 21 Sep 2007 23:32:14 +0200

initramfs-tools-tcos (0.83.2) unstable; urgency=low

  * bin/configurexorg, hooks-addons/40etc3800:
    - Support new openchrome driver
  * scripts/tcos-top/20mountnfs:
    - Mount NFS without nfsro (deprecated in new unionfs)
  * scripts/tcos-top/50fstab:
    - Call udevtrigger and udevsettle to create ide/block devices
  * scripts/tcos-premount/25nbd_filesystem:
    - Create only 2 nbd devices and wait for /dev/nbd0

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 19 Sep 2007 22:41:27 +0200

initramfs-tools-tcos (0.83.1) unstable; urgency=low

  * scripts/tcos-bottom/05inittcos, conf/tcos.conf
    - limit MAX_MEM used for thin clients with ulimit (default 95%)
  * Remove PATH definition in all scripts
  * conf/tcos-run-functions.sh:
    - Disable change vt in kill_usplash
  * scripts/tcos-top/50fstab:
    - Add option to force not use swap
    - Replace cut with awk
  * scripts/tcos-bottom/10ldconfig, hooks/tcosbins:
    - Move ldconfig to /usr/sbin
  * bin/gentcos:
    - Fix typo in inst_pkg function
  * hooks-addons/08discover:
    - Move discover /lib files to squashed /usr/lib (less ram)

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri, 14 Sep 2007 21:32:49 +0200

initramfs-tools-tcos (0.82.11) unstable; urgency=low

  * Add USB/IP experimental support

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 05 Sep 2007 23:06:21 +0200

initramfs-tools-tcos (0.82.10) unstable; urgency=low

  * Update debian unstable kernel version 2.6.22-2-486

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 05 Sep 2007 22:04:44 +0200

initramfs-tools-tcos (0.82.9) unstable; urgency=low

  * bin/gentcos, hooks-addons/50xorg:
    - Add -fix-nvidia-ati option to cache xserver-xorg-core and libgl1-mesa-glx
    - Reorganize bin/gentcos with simpler code
  * bin/tcos-gdm-autologin:
    - Remove wrong licence line (thanks to Quiran to alert me)
  * daemonize/daemon.c, daemonize/Makefile:
    - Make work daemonize with double fork.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 04 Sep 2007 11:10:17 +0200

initramfs-tools-tcos (0.82.8) unstable; urgency=low

  * Add support for cdaudio (using cdfs + cdparanoia to detect)
  * Update dapper kernel 2.6.15-29-386

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 01 Sep 2007 13:18:30 +0200

initramfs-tools-tcos (0.82.7) unstable; urgency=low

  * bin/get_filesystem:
    - Remove disktype
    - Add fstype (new klibc)
  * Update Ubuntu gutsy kernel version

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri, 24 Aug 2007 17:59:18 +0200

initramfs-tools-tcos (0.82.6) unstable; urgency=low

  * bin/tcos-gdm-autologin:
    - New option (--noaction) for debug proposes only
    - If DISPLAY contains IP address instead of hostname try to convert using gethostbyaddr
    - If /etc/gd/gdm.conf-custom exists use it.
    - Update man page
  * conf/tcos-run-functions.sh:
    - Kill tryXorg too in kill_xorg()
    - Kill process in background

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 21 Aug 2007 23:10:18 +0200

initramfs-tools-tcos (0.82.5) unstable; urgency=low

  * Add Subversion repository URL to debian/control.
  * bin/gentcos:
    - Ignore errors in ramdisk size

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 20 Aug 2007 18:28:27 +0200

initramfs-tools-tcos (0.82.4) unstable; urgency=low

  * Add support for lp_server (if avalaible)
  * Change Debian etch kernel version to 2.6.18-5-486

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 20 Aug 2007 11:08:59 +0200

initramfs-tools-tcos (0.82.3) unstable; urgency=low

  * Update kernel versions (enable update && security mirrors in pbuilder)

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 07 Aug 2007 13:32:34 +0200

initramfs-tools-tcos (0.82.2) unstable; urgency=low

  * Fix new gentcos test about unionfs|aufs module (crash without awk)
  * Create /sbin/startesd always not only with cached esound

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 30 Jul 2007 23:39:09 +0200

initramfs-tools-tcos (0.82.1) unstable; urgency=low

  * Add printer support TCOS_PRINT
  * Add initial NBD support (don't work fully yet)
  * Add support for using aufs module instead of unionfs

xmlrpc/tcospasswd.c
/*# tcospasswd.c return an crypt passwd 2006-09-09 14:22:40 mariodebian $
#
# This file is part of tcosxmlrpc.
#
# tcosxmlrpc is free software; you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# tcosxmlrpc is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with tcosxmlrpc; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
# USA.
*/
#include <stdio.h>
#include <string.h>


#include <pwd.h>
#include "validate.h"

void usage();

int main( int argc, char **argv) {

 char *pass;
 if (argc==2) {
   printf("%s\n", crypt(argv[1], PASS_ID) );
   return 0;
 }

 if (argc != 2) {
   pass=getpass("TCOS Password: ");
   printf("%s\n", crypt(pass, PASS_ID) );
   return 1;
 }

 usage();
 return 1;
}


void
usage()
{
  printf("Usage:\n\t\t tcospasswd pass\n");
}


